lang/python/python3-readline/patches/010-link-with-system-libs.patch

```diff
@@ -0,0 +1,78 @@
+Links dynamically to system's libreadline/libhistory.
+--- a/setup.py
++++ b/setup.py
+@@ -7,11 +7,6 @@ import subprocess
+ 
+ from setuptools import setup, Extension
+ 
+-if sys.platform == 'win32':
+-    sys.exit('Error: this module is not meant to work on Windows (try pyreadline instead)')
+-elif sys.platform == 'cygwin':
+-    sys.exit('Error: this module is not needed for Cygwin (and probably does not compile anyway)')
+-
+ here = os.path.abspath(os.path.dirname(__file__))
+ README = open(os.path.join(here, 'README.rst')).read()
+ NEWS = open(os.path.join(here, 'NEWS.rst')).read()
+@@ -47,7 +42,6 @@ DEFINE_MACROS = [
+     ('HAVE_RL_RESIZE_TERMINAL', None),
+ ]
+ 
+-
+ def which_shell():
+     valid_paths = ["/bin/bash", "/usr/local/bin/bash", "/bin/sh"]
+     for path in valid_paths:
+@@ -55,33 +49,18 @@ def which_shell():
+             return path
+     raise IOError("No Shell Found")
+ 
+-
+ # Check if any of the distutils commands involves building the module,
+ # and check for quiet vs. verbose option
+ building = False
+ verbose = True
+ for s in sys.argv[1:]:
+-    if s.startswith('bdist') or s.startswith('build') or s.startswith('install'):
+-        building = True
++#    if s.startswith('bdist') or s.startswith('build') or s.startswith('install'):
++#        building = True
+     if s in ['--quiet', '-q']:
+         verbose = False
+     if s in ['--verbose', '-v']:
+         verbose = True
+ 
+-# Build readline first, if it is not there and we are building the module
+-if building and not os.path.exists('readline/libreadline.a'):
```

Comment:
if this must be removed, an idea would be to simplify this patch by doing:
```
-if building and not os.path.exists('readline/libreadline.a'):
+if False:
```

over-time, these patches need to be re-applied [when upgrading the package] and they do get annoying to re-apply when there's big chunks of code;

Reply:
Agreed